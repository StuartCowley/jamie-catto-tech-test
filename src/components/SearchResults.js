import React from 'react';
import '../styles/searchResults.css';
import PropTypes from 'prop-types';

const SearchResults = ({ results }) => {
    if (!results.length) {
        return <p className='no-results'>No results.</p>
        
    }
    return (
        <div className='search-results'>
            {results.map((img, index) => {
                return (
                    <img 
                    className="search-result-item" 
                    src={img}
                    alt=""
                    key={index}
                    data-testid="result-item" />
                    )
                })}
         </div>
    );
}

SearchResults.propTypes = {
    results: PropTypes.arrayOf(PropTypes.string),
}

export default SearchResults;