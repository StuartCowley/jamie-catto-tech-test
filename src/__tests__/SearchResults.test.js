import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const validProps = [
    'image 1',
    'image 2'
  ];

  const { asFragment } = render(<SearchResults results={validProps}/>);


  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders only text if there are no results', () => {
    const invalidProps = [];

    render(<SearchResults results={invalidProps} />);

    expect(screen.getByText('No results.')).toBeInTheDocument();
  });

  it('renders correct amount of images if there are results', () => {
    render(<SearchResults results={validProps} />);

    expect(screen.getAllByTestId('result-item')).toHaveLength(2);
  });
});