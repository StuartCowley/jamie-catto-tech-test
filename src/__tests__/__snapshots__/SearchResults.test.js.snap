// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <img
    alt=""
    class="search-result-item"
    data-testid="result-item"
    src="image 1"
  />
  <img
    alt=""
    class="search-result-item"
    data-testid="result-item"
    src="image 2"
  />
</DocumentFragment>
`;
